### ISSUE TITLE
`make install` does not nuke existing completions

### ISSUE THREAD

**Issue by mqudsi** [fish-shell/fish-shell#4314]
0ebff8c516c9755d6d7da60067f8d13013018a40 removes the custom completion for the `source` builtin, but it turns out that `make install` over an existing fish installation does not remove the existing completions. Would this be considered a bug or do we expect users to first run `make uninstall` before running `make install`?

Is there a point in trying to sync the two directories instead of nuking the existing completions directory and copying everything over (for installation speed purposes)? (I wouldn't spend much brain power over this one, just `rsync` if `rsync` exists or nuke and copy.)

**Comment by krader1961** [comment]
Good catch. That should definitely be improved if only to keep developers from pulling their hair out after doing `make install` and getting unexpected behavior. It doesn't affect the vast majority of users who install from packages or via mechanisms like Homebrew. So low priority to fix but it would be nice to improve this.

**Comment by zanchey** [comment]
If you run `make install` without blowing away the existing installation or using `make uninstall`, you're gonna have a bad time.

**Comment by mqudsi** [comment]
@zanchey can we check if $PREFIX/bin/fish exists and have make call `make uninstall` from within the `install` target? Or is that too unorthodox?

**Comment by Darkshadow2** [comment]
I just delete *$__fish_datadir/functions* and *$__fish_datadir/completions* before I do `make install`. Is there anything else that really needs to be done other than that?

**Comment by zanchey** [comment]
@mqudsi, that might work - something like:

```Make
ifeq "$(strip $(wildcard $(DESTDIR)$(bindir)/fish))" "$(DESTDIR)$(bindir)/fish"
install: uninstall
endif
```

**Comment by zanchey** [comment]
@Darkshadow2, occasionally you will get bitten by binaries appearing or disappearing - the `check_legacy_binaries target` _should_ take care of that, though.

**Comment by zanchey** [comment]
Fixed with embed-files
